Runtime/PlayableAssets/StreamingImageSequence/StreamingImageSequencePlayableMixer.cs

```diff
@@ -158,10 +160,37 @@ public override void ProcessFrame(Playable playable, FrameData info, object play
                 if (sprite.texture != tex) {
                     m_spriteRenderer.sprite = Sprite.Create(tex, new Rect(0.0f, 0.0f, tex.width, tex.height), new Vector2(0.5f, 0.5f), 100.0f, 2, SpriteMeshType.FullRect);
                 }
-            } else if (null!=m_meshRenderer) {
-                Material mat = m_meshRenderer.sharedMaterial;
-                mat.mainTexture = tex; 
-            } else if (null!= m_image) {
+            } else if (null!=m_meshRenderer)
+            {
+                Material mat;
+                Debug.Log(m_meshRenderer.sharedMaterial + "single material");
+                if (m_meshRenderer.sharedMaterials.Length > 1)
+                {
+                    Debug.Log(m_meshRenderer.sharedMaterial + "masterial");
+                    mat = m_meshRenderer.sharedMaterials[2];
+                }
+                else
+                {
+                    
+                    mat = m_meshRenderer.sharedMaterial;
+                }
+                //mat.mainTexture = tex; 
+                mat.SetTexture("_GraphicsMap",tex);
+            } 
+            else if (null!=m_skinnedmeshRenderer)
+            {
+                Material mat;
+                if (m_skinnedmeshRenderer.sharedMaterials.Length > 1)
+                {
+                    mat = m_skinnedmeshRenderer.sharedMaterials[2];
+                }
+                else
+                {
+                    mat = m_skinnedmeshRenderer.sharedMaterial;
+                }
+                //mat.mainTexture = tex; 
+                mat.SetTexture("_GraphicsMap",tex);
```

Comment:
note: this swaps the _MainTexture from builtin renderer to HDRP's _GraphicsMap texture - need a better way to handle the texture input mapping

Reply:
Could you please make a function to avoid duplicating code with m_meshRenderer above ?
Also, since this only applies to HDRP, perhaps we can use versionDefines and ifdef combination ?

Reply:
the ideal solution would be to have maybe some presets and/or a way to map to any material property, then we don't need to have dependencies or defines - the default is maybe the built in _MainTex property, but users can override in the ImageSequence component?

Will see if I can rework this

Reply:
That sounds like a good idea, but will probably need a much bigger change, and we might rework it soon too.

Your changes now looks good, but I'd like to have it work not just in HDRP. At least legacy should be supported.

Reply:
will wrap with conditional defines so that it supports builtin and hdrp - not sure what the equivalent urp texture channels would be, but would be easy enough to add as well

Reply:
Ok, it would be great if you can add URP support as well.